modules/chargebee-integration/app/kuberlogic.go

```diff
@@ -41,7 +44,124 @@ func createService(logger *zap.SugaredLogger, svc *models.Service) error {
 	return nil
 }
 
+func deleteService(logger *zap.SugaredLogger, serviceId string) error {
+	params := service.NewServiceDeleteParams()
+	params.ServiceID = serviceId
+
+	apiClient, err := makeClient()
+	if err != nil {
+		return err
+	}
+	_, err = apiClient.Service.ServiceDelete(
+		params,
+		httptransport.APIKeyAuth("X-Token", "header", viper.GetString("KUBERLOGIC_APISERVER_TOKEN")),
+	)
+	return nil
+}
+
+func getServiceBySubscriptionId(logger *zap.SugaredLogger, subscriptionId string) (*models.Service, error) {
+	params := service.NewServiceListParams()
+	params.SubscriptionID = &subscriptionId
+
+	apiClient, err := makeClient()
+	if err != nil {
+		return nil, err
+	}
+
+	response, err := apiClient.Service.ServiceList(
+		params,
+		httptransport.APIKeyAuth("X-Token", "header", viper.GetString("KUBERLOGIC_APISERVER_TOKEN")),
+	)
+	if err != nil {
+		return nil, err
+	}
+	if len(response.Payload) > 0 {
+		return response.Payload[0], nil
+	}
+	return nil, errors.Errorf("There is no service with subscription id: %s", subscriptionId)
+}
+
+func addServiceBackup(logger *zap.SugaredLogger, serviceID string) (*models.Backup, error) {
+	params := backup.NewBackupAddParams()
+	params.BackupItem = new(models.Backup)
+	params.BackupItem.ServiceID = serviceID
+
+	apiClient, err := makeClient()
+	if err != nil {
+		return nil, err
+	}
+	response, err := apiClient.Backup.BackupAdd(
+		params,
+		httptransport.APIKeyAuth("X-Token", "header", viper.GetString("KUBERLOGIC_APISERVER_TOKEN")),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return response.Payload, nil
+}
+
+func listServiceBackups(logger *zap.SugaredLogger, serviceID string) (models.Backups, error) {
+	params := backup.NewBackupListParams()
+	params.ServiceID = &serviceID
+
+	apiClient, err := makeClient()
+	if err != nil {
+		return nil, err
+	}
+	response, err := apiClient.Backup.BackupList(
+		params,
+		httptransport.APIKeyAuth("X-Token", "header", viper.GetString("KUBERLOGIC_APISERVER_TOKEN")),
+	)
+	if err != nil {
+		return nil, err
+	}
+	return response.Payload, nil
+}
+
+func deleteServiceBackup(logger *zap.SugaredLogger, backupID string) error {
+	params := backup.NewBackupDeleteParams()
+	params.BackupID = backupID
+
+	apiClient, err := makeClient()
+	if err != nil {
+		return err
+	}
+	_, err = apiClient.Backup.BackupDelete(
+		params,
+		httptransport.APIKeyAuth("X-Token", "header", viper.GetString("KUBERLOGIC_APISERVER_TOKEN")),
+	)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func waitForBackup(logger *zap.SugaredLogger, serviceID string, backupID string) error {
+	timeout := time.Second
+	for i := 13; i > 0; i-- {
```

Comment:
13 - magic numbers. Is it retries? Could you add comments or rename to constant?